versioncontrol/versioncontrol.go

```diff
@@ -0,0 +1,10 @@
+package versioncontrol
+
+import "net/http"
+
+//Client holds the necessary items to make api calls to various version control providers
+//TODO Would be nice to attach repo urls and paths to this object at some point
+type Client struct {
```

Comment:
I would move this struct and its methods in with the generator. I believe that is the only place this is used.

Reply:
Then it would be a `report.Client` which makes sense to me. 

Otherwise a `versioncontrol.Client` is also fine. 

Up to you.

Reply:
Go does not allow you to define a receiver method outside of the package which defines the receiver. Since we have methods defined on Client in the version control package, we can't simply move it to the report package without further refactoring. 

I set this up with an eye to future additions of urls for each version control provider. Once that improvement is in place, this will make much more sense. I would also want to consider renaming it to something more like "RepoFetcher" to make it's function more clear. Happy to discuss further if you would like.

https://github.com/1Password/dep-report/blob/kb/support-go-modules/versioncontrol/github.go#L84
https://github.com/1Password/dep-report/blob/kb/support-go-modules/versioncontrol/gerrit.go#L113

Reply:
Yep I was thinking the get methods would go in the same package as the client.

Aka make a new receiver package, and move all receiver methods in with it, including the two you linked. Then you get all the functions in the same package, which imo allows for a clearer view to refactoring out repo specific behaviour. 

Again this is just a suggestion.